src/pke/lib/scheme/bfvrns/bfvrns-parametergeneration.cpp

```diff
@@ -160,15 +173,15 @@ bool ParameterGenerationBFVRNS::ParamsGenBFVRNS(std::shared_ptr<CryptoParameters
 
         // this code updates n and q to account for the discrete size of CRT moduli
         // = dcrtBits
-        int32_t k = static_cast<int32_t>(std::ceil((std::ceil(logq / std::log(2)) + 1.0) / dcrtBits));
+        int32_t k = static_cast<int32_t>(std::ceil(std::ceil(logq) / dcrtBits));
 
-        double logqCeil = k * dcrtBits * std::log(2);
+        double logqCeil = k * dcrtBits;
 
         while (nRLWE(logqCeil) > n) {
             n        = 2 * n;
             logq     = logqBFV(n);
-            k        = static_cast<int32_t>(std::ceil((std::ceil(logq / std::log(2)) + 1.0) / dcrtBits));
-            logqCeil = k * dcrtBits * std::log(2);
```

Comment:
Not adding the +1 anymore? The same comment for the occurrences below.

Reply:
I don't see why + 1 is needed in the current version of OpenFHE. Maybe in an old version (of PALISADE), we sometimes had larger moduli (NextPrime might have been used). So in my mind, ceil(logq/dcrtBits) is sufficient and guarantees that that logq<=k*dcrtBits. Moreover, adding + 1 would cause problems in the new estimation logic (getting an extra RNS limb when it is not needed).